### ISSUE TITLE
`quotes` does not report template literals that contain a newline escape character

### ISSUE THREAD

**Issue by not-an-aardvark** [eslint/eslint#7646]
**Tell us about your environment**

* **ESLint Version:** master
* **Node Version:** 7.2.0
* **npm Version:** 3.10.9

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

(none)

**What did you do? Please include the actual source code causing the issue.**

```js

/* eslint quotes: [error, "single"] */

var foo = `foo\nbar`;

```

**What did you expect to happen?**

ESLint should report an error, and be able to fix the line to:

```js
var foo = 'foo\nbar';
```

**What actually happened? Please include the actual, raw output from ESLint.**

No errors were reported.

I'm guessing this is a bug in the logic that detects multiline template literals. The rule does not report lines like this (as expected):

```js
var foo = `foo
bar`; // No error (working as intended)
```

But it should still report lines that literally contain `\n`.
